src/Types/UserGroup.elm

```diff
@@ -0,0 +1,285 @@
+module Types.UserGroup exposing (..)
+
+import Json.Decode as Decode
+import Json.Encode as Encode
```

Comment:
NIT: Given you created Codec, maybe using Codec package from minibill/elm-codec will be nice touch here.